infrastructure/tooling/src/generate/generators/go-version.js

```diff
@@ -67,15 +70,24 @@ exports.tasks = [{
         `MIN_GO_MINOR_VERSION=${goVersionMinor}`));
 
     utils.status({ message: 'workers/generic-worker/gw-decision-task/tasks.yml' });
+    const previousGoVersion = (await readRepoFile('workers/generic-worker/gw-decision-task/tasks.yml')).match(/go[0-9.]+/)[0].substring(2).split('.');
```

Comment:
maybe we need some better way of storing current version? 
Right now if I do same match on the file it gives me 19 matches, where `go.(mod)` and `go.(sum)` are also being part of those

Or at least maybe update regexp to be more predictable and less greedy with that dot: `/go(([0-9]+\.?)+)/g` 

```
'abc go. | go123.123 | go.sum | go1.20.3 | go1.18'.match(/go(([0-9]+\.?)+)/g)
--> 
[ "go123.123", "go1.20.3", "go1.18" ]
```